lib/views/course_view/downloaded_pinned_courses_view/pinned_courses_view.dart

```diff
@@ -1,29 +1,68 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:gocast_mobile/providers.dart';
+import 'package:gocast_mobile/views/components/base_view.dart';
 import 'package:gocast_mobile/views/course_view/downloaded_pinned_courses_view/content_view.dart';
 import 'package:gocast_mobile/views/video_view/video_card_view.dart';
 
-/// PinnedCourses
-///
-/// A widget representing the 'Pinned Courses' section of the app.
-/// It utilizes the CourseContentScreen to display a list of courses
-/// that the user has pinned.
-class PinnedCourses extends ConsumerWidget {
+class PinnedCourses extends ConsumerStatefulWidget {
   const PinnedCourses({super.key});
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    return const CourseContentScreen(
-      title: 'Pinned',
-      videoCards: [
-        VideoCard(
-          imageName: 'assets/images/course2.png',
-          title: 'Computer Science [CS202]',
-          date: 'July 23, 2019',
-          duration: '02:00:00',
-        ),
-        // Add more VideoCard widgets as needed
-      ],
+  PinnedCoursesState createState() => PinnedCoursesState();
+}
+
+class PinnedCoursesState extends ConsumerState<PinnedCourses> {
+  @override
+  void initState() {
+    super.initState();
+    Future.microtask(
+        () => ref.read(userViewModelProvider.notifier).fetchUserPinned(),);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final userPinned = ref.watch(userViewModelProvider).userPinned ?? [];
+
+    return Scaffold(
+      body: RefreshIndicator(
+        onRefresh: () async {
+          await ref.read(userViewModelProvider.notifier).fetchUserPinned();
+        },
+        child: userPinned.isNotEmpty
+            ? CourseContentScreen(
+                title: "Pinned",
+                videoCards: userPinned.map((course) {
+                  return VideoCard(
+                    imageName: 'assets/images/course2.png',
+                    title: "${course.name} - ${course.slug}",
+                    date:
+                        "${course.semester.year} ${course.semester.teachingTerm}",
+                    duration: course.cameraPresetPreferences,
+                    onTap: () {},
+                  );
+                }).toList(),
+                onRefresh: () async {
```

Comment:
The following lines (45-49) can be removed, since reloading is already implemented in lines 28-31.
```dart
onRefresh: () async {
  await ref
      .read(userViewModelProvider.notifier)
      .fetchUserPinned();
},
```

Reply:
✅